### ISSUE TITLE
yarn upgrade --exact doesn't work as expected

### ISSUE THREAD

**Issue by npasquet** [yarnpkg/yarn#4244]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

It prepend `^` to package version

**If the current behavior is a bug, please provide the steps to reproduce.**

*package.json before upgrade command*
```
devDependencies: {
  "@storybook/addon-actions": "3.2.0",
   "@storybook/addon-centered": "3.2.0",
   "@storybook/addon-info": "3.2.0",
   "@storybook/react": "3.2.3",
}
```

Run `yarn upgrade @storybook/react @storybook/addon-actions @storybook/addon-centered @storybook/addon-info --latest --exact`

*package.json after upgrade command*
```
devDependencies: {
    "@storybook/addon-actions": "^3.2.6",
    "@storybook/addon-centered": "^3.2.7",
    "@storybook/addon-info": "^3.2.7",
    "@storybook/react": "^3.2.8",
}
```

**What is the expected behavior?**

It should respect `--exact` option.

**Please mention your node.js, yarn and operating system version.**
node: v8.1.0
yarn: v0.27.5
macOs Sierra: v10.12.6

**Comment by rally25rs** [comment]
Yikes! It looks like the documentation for changes to `upgrade` made it out the door before the actual code. The `--exact` flag was added in #3847 but that code has not yet been released.

In Yarn v0.27.5 `--exact` exists for `upgrade-interactive` but not `upgrade`. In the above PR I added the flag for `upgrade` as well so that the commands would have consistent arguments.

The changed code hasn't been tagged yet, but I think this will be in Yarn v0.28.5, or v1 if we jump right to that.

We probably need a better way of tying doc website changes directly to code changes when they go live :( I apologize for the confusion. I'll open some discussion around keeping the docs inline with the latest Yarn release over in our development Slack channel.

I'm also going to close this issue since the code is there to make it work, and just waiting for the next release.
